[KB-107] Write problem definition draft (#85)
chhoumann/thesis-chemcam: report_thesis/src/sections/problem_definition.tex

> \section{Problem Definition}

Reviewer: Generally this section is really good. However, I am not sure whether or not the introduction to this section fits here. 
That said, it may. I think we should have a discussion about it and perhaps look into what other papers do.

My reason for questioning it is purely based on my understanding that this section should be almost purely "mathy" and as such a lot of the things explained in the introduction should be elsewhere.

Reviewer: Following this logic, the same could perhaps be said for the ending as well. Maybe this is more of a problem statement? Let's discuss it.

Author: See my message in our chat. Not relevant here. But yes, let's talk abuot it.

\label{sec:problem_definition}

> Our research aims to improve the accuracy and robustness of major oxide predictions derived from \gls{libs} data, building upon the baseline established in \cite{p9_paper}.

Reviewer 2: ```suggestion
Our work aims to improve the accuracy and robustness of major oxide predictions derived from \gls{libs} data, building upon the baseline established in \citet{p9_paper}.
```

> There are many challenges in predicting major oxides from \gls{libs} data, including the high dimensionality and non-linearity of the data, as well as the presence of multicollinearity.

Reviewer 2: Citation needed?

> Some of these are caused by \textit{matrix effects}\cite{andersonImprovedAccuracyQuantitative2017}, which is a catch-all term for any effect that can cause the intensity of emission lines from an element to vary, independent of that element's concentration. So it's unknown variables that affect the results.

Reviewer 2: ```suggestion
Some of these are caused by \textit{matrix effects}\cite{andersonImprovedAccuracyQuantitative2017}, which is an umbrella term for any effect that can cause the intensity of emission lines from an element to vary, independent of that element's concentration. 
In other words, unknown variables that affect the results.
```

Author: That's the point

Furthermore, due to the high cost of data collection, datasets are often small, which further complicates the task of building accurate and robust models.

> Based on the limitations with the current MOC pipeline, as reported in \cite{p9_paper}, we identified three key areas for further investigation: dimensionality reduction, model selection, and outlier removal.

Reviewer 2: ```suggestion
Based on the limitations with the current \gls{moc} pipeline, as reported in \cite{p9_paper}, we identified three key areas for further investigation: dimensionality reduction, model selection, and outlier removal.
```

In this work, we focus on dimensionality reduction and model selection over outlier removal.

> This is justified by the low incidence of outliers in the ChemCam \gls{libs} calibration dataset, as reported in \cite{p9_paper}.

Reviewer 2: ```suggestion
This is justified by the low incidence of outliers in the \gls{chemcam} \gls{libs} calibration dataset, as reported in \citet{p9_paper}.
```

Dimensionality reduction is crucial for managing the high-dimensional nature of \gls{libs} data. % TODO: There are lots of related works which explore DR in LIBS data. We can back this up with citations.

> Furthermore, model selection shows promise in addressing the limitations of the current MOC pipeline, as it allows for the exploration of a wider range of algorithms, potentially leading to improved performance.

Reviewer 2: ```suggestion
Furthermore, model selection shows promise in addressing the limitations of the current \gls{moc} pipeline, as it allows for the exploration of a wider range of algorithms, potentially leading to improved performance.
```

> We showed that advanced ensemble methods, such as \gls{gbr}, and deep learning models, such as \gls{ann}, have the potential to outperform the current MOC pipeline.

Reviewer 2: ```suggestion
We showed that advanced ensemble methods like \gls{gbr} and deep learning models like \gls{ann} have the potential to outperform the current \gls{moc} pipeline.
```

Thought the sentence structure was a little heavy.

Author: ```suggestion
We showed that advanced ensemble methods like \gls{gbr} and deep learning models like \gls{ann}s have the potential to outperform the current \gls{moc} pipeline.
```

> Methods are selected based on their promise in handling high-dimensional, non-linear data. Ideally, the selected methods should also be feasible for small datasets, a common scenario in \gls{libs} analyses. 

Reviewer 2: Their "promise"? What does that mean? Do you mean "potential to handle high-dimensional, non-linear data"?

Author: Promise can be used analogously to potential in this case.

> In order to address the aforementioned challenges, we propose to explore advanced ensemble methods and deep learning models, which have shown promise in handling high-dimensional, non-linear data.

Reviewer 2: Also needs a cite.

> In order to address the aforementioned challenges, we propose to explore advanced ensemble methods and deep learning models, which have shown promise in handling high-dimensional, non-linear data.

Author: No need to restate it. Cites aren't relevant at this point; please see the task description.

It is necessary to establish metrics to evaluate the performance of the models.
In \cite{p9_paper}, we proposed to use the \gls{rmse} as a proxy for accuracy.

\gls{rmse} is given by:

\begin{equation}
    RMSE = \sqrt{\frac{1}{n} \sum_{i=1}^{n} (y_i - \hat{y}_i)^2}
\end{equation}

where $y_i$ represents the actual values, $\hat{y}_i$ the predicted values, and $n$ the number of observations.

To address robustness, we propose considering the standard deviation of prediction errors across each oxide and test instance, defined as:

\begin{equation}

>     \sigma_{error} = \sqrt{\frac{1}{n-1} \sum_{i=1}^{n} (e_i - \bar{e})^2}

Reviewer 2: ```suggestion
    \sigma_{error} = \sqrt{\frac{1}{n-1} \sum_{i=1}^{n} (e_i - \bar{e})^2},
```
Same as before

\end{equation}

where $e_i = y_i - \hat{y}_i$ and $\bar{e}$ is the mean error.

In order to narrow down the scope of our research, we set the following constraints:
\begin{itemize}

>     \item Prioritize normalization across individual spectrometers' wavelength ranges over full-spectrum normalization.

Reviewer 2: Should we mention why we do this? And maybe reference SuperCam? The reasoning behind this constraint isn't trivially obvious to someone in the CS field.

Author: For style reasons, I prefer the bullets being short points that I expand upon after. This is rather common.

Author: Added clarification

    \item Focus on techniques proven effective for non-linear, high-dimensional data, even outside the \gls{libs} context.
    \item Ensure methods are feasible for small datasets.
\end{itemize}

In \cite{p9_paper}, we used both full-spectrum normalization (Norm 1) and normalization across individual spectrometers' wavelength ranges (Norm 3).

> However, in this work, we opt for normalizing across individual spectrometers' wavelength ranges.

Reviewer 2: ```suggestion
However, in this work, we opt to always normaliz across individual spectrometers' wavelength ranges (Norm 3).
```

> This decision is guided by the operational parameters of SuperCam\cite{andersonPostLandingMajorElements2022}, where they do not normalize across the entire spectrum, but rather across individual spectrometers' wavelength ranges.

Reviewer 2: What is meant by "operational parameters"? Should maybe elaborate a bit since this is a fairly major decision.

Author: Clarified

> In order to ensure the future applicability of our methods, we follow the same normalization approach.

Reviewer 2: ```suggestion
To ensure the future applicability of our methods and to align our work with NASA's standards, we follow the same normalization approach.
```

I think the reasoning goes beyond just future applicability - we do also want to align ourselves with NASA's standards if we use their data, no?

Author: `NASA's standards` doesn't really make sense. I've clarified the sentence, though.

Our methodologies are selected to ensure compatibility with small datasets, a common scenario in \gls{libs} analyses.
This consideration informs our preference for cross-validation techniques, which offer robust performance assessments without necessitating large data volumes.

> Additionally, the computational complexity of models is deemed a secondary concern, given the scale of data typically involved in \gls{libs} studies, allowing for a broader exploration of sophisticated, potentially more computationally intensive algorithms.

Reviewer 2: This seems to come out of the blue, however I realize that it's supposed to elaborate on the constraints listed above. This goes back to my previous comment - wouldn't it be best to elaborate on each constraints in-situ instead of further down? This would make it much easier to read, and we wouldn't be jumping back and forth so much. I propose listing a constraint, then immediately elaborate on the reasoning for this constraint before listing the next constraint.

Author: I don't see the comment you reference, but I've decided to simply delete this part. It's more relevant in the methodology section.
